src/main/cpp/lang/js.cpp

```diff
@@ -214,6 +214,95 @@ std::size_t match_hashbang_comment(std::u8string_view s) noexcept
     return length;
 }
 
+std::size_t match_escape_sequence(std::u8string_view str) noexcept
+{
+    // https://262.ecma-international.org/15.0/index.html#prod-EscapeSequence
+    // All escape sequences must start with backslash.
+    if (str.empty() || str[0] != u8'\\') {
+        return 0;
+    }
+
+    // Need 1 more char after the backslash.
+    if (str.length() < 2) {
+        return 1;
+    }
```

Comment:
A single backslash shouldn't be considered an escape sequence if it's not followed by something that can be escaped.
```suggestion
    if (str.length() < 2 || str[0] != u8'\\') {
        return 0;
    }
```